pkg/k8sutil/pv.go

```diff
@@ -75,12 +75,22 @@ func UpdateReclaimPolicy(client *k8s.Clientset, pv *corev1.PersistentVolume) err
 	return err
 }
 
-func GetFlexVolumeName(pv *corev1.PersistentVolume) string {
+func GetVolumeName(pv *corev1.PersistentVolume) string {
 	// Rook creates rbd image with PV name
 	return pv.Name
 }
 
 func GetCSIVolumeName(pv *corev1.PersistentVolume) string {
+	retry := 0
+	maxRetry := 5
+	for pv.Spec.CSI.VolumeAttributes["imageName"] == "" && retry < maxRetry {
```

Comment:
The value of `pv.Spec.CSI.VolumeAttributes` is never going to change in this loop. You would need to retrieve the PV again from the API.

Reply:
Thanks for catching it! Updated to fetch value at each iteration.